app/javascript/chat/chat.jsx

```diff
@@ -353,16 +355,20 @@ export default class Chat extends Component {
     });
   };
 
-  setupChannel = (channelId) => {
-    const { messages, messageOffset, activeChannel, appDomain } = this.state;
+  setupChannel = (channelId, activeChannel) => {
+    const { messages, messageOffset, appDomain } = this.state;
     if (
       !messages[channelId] ||
       messages[channelId].length === 0 ||
       messages[channelId][0].reception_method === 'pushed'
     ) {
       getAllMessages(channelId, messageOffset, this.receiveAllMessages);
     }
-    if (activeChannel && activeChannel.channel_type !== 'direct') {
+    if (
```

Comment:
Question: It's not clear to me what this logic is. Is it checking that it's not a DM and that the current channel is indeed the current channel?

Reply:
It is checking the active channel is not DM.